nixos/modules/services/network-filesystems/glusterfs.nix

```diff
@@ -70,11 +128,12 @@ in
         PIDFile="/run/glusterd.pid";
         LimitNOFILE=65536;
         ExecStart="${glusterfs}/sbin/glusterd -p /run/glusterd.pid --log-level=${cfg.logLevel} ${toString cfg.extraFlags}";
-        KillMode="process";
+        KillMode="control-group";
```

Comment:
I would not make this change part of this commit / PR, but do that separately.

I have a branch [nh2-glusterfs-service-improvements](https://github.com/nh2/nixpkgs/tree/nh2-glusterfs-service-improvements) that includes [this commit](https://github.com/nh2/nixpkgs/commit/1cdb0b1d2954b73394a2a17a2b59bdb46360e633) to handle this.

My plan is that after your PR here has landed, we can merge the remaining improvements from my branch, and then do the gluster 3.12 upgrade (#29062).

Reply:
Good point I will remove this.

Reply:
> My plan is that after your PR here has landed, we can merge the remaining improvements from my branch

I've PR'd the changes here now: https://github.com/NixOS/nixpkgs/pull/29868